internal/circle/errors.go

```diff
@@ -28,15 +32,33 @@ func (e APIError) Error() string {
 	return fmt.Sprintf("APIError: Code=%d, Message=%s, Errors=%v", e.Code, e.Message, e.Errors)
 }
 
+type TenantStatusUpdater struct {
+	tenantManager tenant.ManagerInterface
+}
+
+var invalidAPIKeyStatusCodes = []int{http.StatusUnauthorized, http.StatusForbidden}
```

Comment:
nit: there are other *invalid* status codes like 5xx, so we should rename this variable to something like `authErrorStatusCodes` to be more accurate.